### ISSUE TITLE
»Console has to be executed with web server user« offers no direct help

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/core#21413]
```
$ sudo php console.php files:scan --all
Console has to be executed with the user that owns the file config/config.php
Current user: root
Owner of config.php: www-data
```
Shouldn’t it be possible to run that as root as well, since that user has access to all files anyway?

Because right now, this restriction led me to defining this ridiculous alias:
```
alias owncloudscan='cd ~/owncloud; sudo chown root config/config.php; sudo php console.php files:scan --all; sudo chown www-data config/config.php'
```

**Comment by RealRancor** [comment]
@jancborchardt Why not running it as the webservers user as asked? See also the documentation:

https://doc.owncloud.org/server/8.2/admin_manual/configuration_server/occ_command.html#run-occ-as-your-http-user

**Comment by jancborchardt** [comment]
Right, thanks for the tip!

Nevertheless, now I get these strange errors:
```
$ sudo -u www-data php occ files:scan --all
Home storage for user heinz not writable
Make sure you're running the scan command only as the user the web server runs as
Home storage for user thomas not writable
Make sure you're running the scan command only as the user the web server runs as
```
They didn’t show up before.

**Comment by RealRancor** [comment]
Tried to set the correct permissions? https://doc.owncloud.org/server/8.2/admin_manual/installation/installation_wizard.html#setting-strong-directory-permissions

**Comment by MorrisJobke** [comment]
> They didn’t show up before.

Yep. Because you forced the script to run as root you messed up your directory permissions and the web server can't read/write anymore (because some files where written by root). Nothing we can do here. If the admin decides to force trick our own checks (we added this because of this specific problem you now run into). 

Nothing we can do here - try to give the web server the permissions to read the files inside data/ and all should be fine.

Closing.

**Comment by jancborchardt** [comment]
Cool, thanks then! :)

**Comment by jancborchardt** [comment]
Also, there’s not »nothing« we can do here. We could simply say in the error message: 

> Use this command to run it as the web server user:
> sudo -u www-data php occ files:scan --all

**Comment by RealRancor** [comment]
Hi,

the main problem with giving such distro specific commands are that it differs from one distro to another. Some don't have sudo installed, some are not using www-data.

**Comment by jancborchardt** [comment]
www-data of course will be taken from the system, just like the line `Owner of config.php: www-data` gets that.

The point here is that the whole error message is not very helpful. We can simply provide the command here instead of requiring people to need to look into the docs.

**Comment by MorrisJobke** [comment]
> We can simply provide the command here instead of requiring people to need to look into the docs.

Not really - As stated above. Not all systems have sudo installed. We can guess a command that could work, but we can't never the sure, that it is the correct command. But we can add the hint, that it needs to be executed as user `www-data` (or what else we detected) and then write down an example command that will work for most setups.

**Comment by RealRancor** [comment]
Exactly, its not possible to give the exact command until the occ can get the used command passed somehow from the command line. 

An alternative would be to just add a text like:

"Add sudo -u $username to the beginning of your command"

as the user already knows the command he had used.

**Comment by jancborchardt** [comment]
Yup, that’s what I mean. It doesn’t have to be the exact command, but the `-u www-data` part can be suggested pretty easily, with the $username gotten from the system.

**Comment by edwardcrompton** [comment]
How about something like this? https://github.com/owncloud/core/pull/21446

**Comment by MorrisJobke** [comment]
Fixed by #21446
